validator_client/src/is_synced.rs

```diff
@@ -0,0 +1,70 @@
+use remote_beacon_node::RemoteBeaconNode;
+use rest_types::SyncingResponse;
+use slog::{debug, error, Logger};
+use types::EthSpec;
+
+/// A distance in slots.
+const SYNC_TOLERANCE: u64 = 4;
+
+/// Returns `true` if the beacon node is synced and ready for action.
+///
+/// Returns `false` if:
+///
+///  - The beacon node is unreachable.
+///  - The beacon node indicates that it is syncing **AND** it is more than `SYNC_TOLERANCE` behind
+///  the highest known slot.
+///
+///  The second condition means the even if the beacon node thinks that it's syncing, we'll still
+///  try to use it if it's close enough to the head.
+pub async fn is_synced<E: EthSpec>(
+    beacon_node: &RemoteBeaconNode<E>,
+    log_opt: Option<&Logger>,
+) -> bool {
+    let resp = match beacon_node.http.node().syncing_status().await {
+        Ok(resp) => resp,
+        Err(e) => {
+            if let Some(log) = log_opt {
+                error!(
+                    log,
+                    "Unable connect to beacon node";
+                    "error" => format!("{:?}", e)
+                )
+            }
+
+            return false;
+        }
+    };
+
+    match &resp {
+        SyncingResponse {
+            is_syncing: false, ..
+        } => true,
+        SyncingResponse {
+            is_syncing: true,
+            sync_status,
+        } => {
+            if let Some(log) = log_opt {
+                debug!(
+                    log,
+                    "Beacon node sync status";
+                    "status" => format!("{:?}", resp),
+                );
+            }
+
+            if sync_status.current_slot + SYNC_TOLERANCE >= sync_status.highest_slot {
```

Comment:
This condition isnt safe. We shouldn't rely on current and highest slot (they are kind of wacky because we can sync multiple head chains at once and the highest slot can be artificially manipulated by a peer). 
The ideal condition would be, slot_clock.saturating_sub(head_slot) < tolerance then consider synced.

Reply:
I've updated this, thanks :)